client/src/components/shared/Cubic.js

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+
+const Cubic = ({ lineIndex, pointIndex, reducers, ...props }) => {
+  return (
+    <g className="ad-Anchor">
+      <line
+        className="ad-Anchor-line"
+        x1={props.p1x}
+        y1={props.p1y}
+        x2={props.x1}
+        y2={props.y1}
+      />
+      <line
+        className="ad-Anchor-line"
+        x1={props.p2x}
+        y1={props.p2y}
+        x2={props.x2}
+        y2={props.y2}
+      />
+      <circle
+        className="ad-Anchor-point"
+        onMouseDown={(e) =>
+          reducers.setDraggedCubic({ lineIndex, pointIndex, anchor: 0 })
+        }
+        cx={props.x1}
+        cy={props.y1}
+        r={1}
+      />
+      <circle
+        className="ad-Anchor-point"
+        onMouseDown={(e) =>
+         reducers.setDraggedCubic({ lineIndex, pointIndex, anchor: 1 })
```

Comment:
One of the `circle` is the handle that modifies the curve, right? 
What is the second `circle`?

Reply:
They are both curve handles. A Cubic Bezier Path has 2 curve modification points. If you run the branch, you will see that there are anchors (which I also call handles sometimes) coming out of both points of the curve. Using your React Dev Tools to inspect the components will allow you too see all their properties, names an indexes, which will make things clearer.